devel/git/Portfile

```diff
@@ -227,35 +228,44 @@ variant credential_osxkeychain description {Install git credential-osxkeychain u
     }
 
     post-destroot {
-        xinstall -m 755 "${worksrcpath}/contrib/credential/osxkeychain/git-credential-osxkeychain" \
+        xinstall -m 0755 \
+            "${worksrcpath}/contrib/credential/osxkeychain/git-credential-osxkeychain" \
             "${destroot}${prefix}/libexec/git-core/"
+    }
 
-        # Create a global gitconfig file that sets credential.helper to
-        # osxkeychain. This can be overriden on a per-user basis in the user's
-        # own git config file
-        set f [open ${destroot}${prefix}/etc/gitconfig w 0644]
-        puts ${f} "\[credential]\n\thelper = osxkeychain"
-        close ${f}
+    pre-activate {
+        set osxkc_config        "\[credential]\n\thelper = osxkeychain"
+
+        # Ensure the osxkeychain credential helper is configured in the
+        # system-wide gitconfig.  If the gitconfig file already exists,
+        # then append the configuration to the pre-existing file only
+        # if the file is currently missing the desired config.
+
+        if { ![file exists ${system_gitconfig}] } {
```

Comment:
You gave a conceptual bug here. When you check the currently existing file you correctly check the *installed* version directly in ${prefix}, but you cannot directly write to this file, which is what you currently do. You have to write to the version in the DESTROOT ie at ${destroot}${prefix}, and if need be first clone the currently installed version before appending to it.